nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/media/ExtractMediaMetadata.java

```diff
@@ -0,0 +1,276 @@
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return this.properties;
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        final String fileNamePatternInput = context.getProperty(FILENAME_FILTER).getValue();
+        if (fileNamePatternInput != null && fileNamePatternInput.length() > 0) {
+            filenameFilterRef.set(Pattern.compile(fileNamePatternInput));
+        } else {
+            filenameFilterRef.set(null);
+        }
+
+        final String mimeTypeFilterInput = context.getProperty(MIME_TYPE_FILTER).getValue();
+        if (mimeTypeFilterInput != null && mimeTypeFilterInput.length() > 0) {
+            mimeTypeFilterRef.set(Pattern.compile(mimeTypeFilterInput));
+        } else {
+            mimeTypeFilterRef.set(null);
+        }
+
+        String metadataKeyFilterInput = context.getProperty(METADATA_KEY_FILTER).getValue();
+        if (metadataKeyFilterInput != null && metadataKeyFilterInput.length() > 0) {
+            metadataKeyFilterRef.set(Pattern.compile(metadataKeyFilterInput));
+        } else {
+            metadataKeyFilterRef.set(null);
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowfile = session.get();
+        if (flowfile == null) {
+            return;
+        }
+
+        // fail fast if filename doesn't match filter
+        Pattern filenameFilter = filenameFilterRef.get();
+        if (filenameFilter != null && !filenameFilter.matcher(flowfile.getAttribute(CoreAttributes.FILENAME.key())).matches()) {
+            session.transfer(flowfile, SUCCESS);
+            return;
+        }
+
+        final ProcessorLog logger = this.getLogger();
+        final ObjectHolder<Map<String, String>> value = new ObjectHolder<>(null);
+        final Integer max = context.getProperty(MAX_NUMBER_OF_ATTRIBUTES).asInteger();
+        final String prefix = context.getProperty(METADATA_KEY_PREFIX).evaluateAttributeExpressions().getValue();
+        final FlowFile ff = flowfile;
+
+        try {
+            session.read(flowfile, new InputStreamCallback() {
+                @Override
+                public void process(InputStream in) throws IOException {
+                    try {
+                        Map<String, String> results = tika_parse(ff, in, prefix, max);
+                        value.set(results);
+                    } catch (SAXException | TikaException e) {
+                        throw new IOException(e);
+                    }
+                }
+            });
+
+            // Write the results to attributes
+            Map<String, String> results = value.get();
+            if (results != null && !results.isEmpty()) {
+                flowfile = session.putAllAttributes(flowfile, results);
+            }
+
+            session.transfer(flowfile, SUCCESS);
```

Comment:
I know the ATTRIBUTES_MODIFIED provenance event is emitted automatically but would be nice to have some additional details added